GeometryDashAPI/Server/Models/MusicInfo.cs

```diff
@@ -1,33 +1,75 @@
 ﻿using System;
 using System.Collections.Generic;
+using System.Globalization;
 using System.Text;
+using GeometryDashAPI.Server.Enums;
 
 namespace GeometryDashAPI.Server.Models
 {
     public class MusicInfo
     {
-        public int ID { get; set; }
-        public string Name { get; set; }
+        #region fields
+        public int ID { get; private set; }
+        public string Name { get; private set; }
+        public string Author { get; private set; }
+        public float Size { get; private set; }
+        public string Link { get; private set; }
+        public OfficialSong OfficialSong { get; private set; } = OfficialSong.NotChoosen;
+
+    public Dictionary<string, string> WithoutLoaded { get; private set; } = new Dictionary<string, string>();
+        #endregion
 
         public MusicInfo()
         {
+
         }
 
         public MusicInfo(string data)
         {
-            string[] splittedData = data.Split(new string[] { "~|~" }, StringSplitOptions.None);
-            for (int i = 0; i < splittedData.Length; i += 2)
+            Load(data);
+        }
+
+        public MusicInfo(OfficialSong officialSong)
+        {
+            this.OfficialSong = officialSong;
+        }
+
+        public void Load(string data)
+        {
+            string[] splitedData = data.Split('|');
+
+            for (int i = 0; i < splitedData.Length - 1; i += 2)
             {
-                switch (splittedData[i])
+                splitedData[i] = splitedData[i].Replace("~", "");
+                String value = splitedData[i + 1].Replace("~", ""); //i mean, should probably do separate variable for each of those functions in Info classes
```

Comment:
Why you use separator as '|' then replacing all '\~' to empty?
I think the separator here is "\~|\~".

Also better to use aliases for variable types.
_string instead of String_.
_int instead of Int32_.

Reply:
> Why you use separator as '|' then replacing all '\~' to empty?
> I think the separator here is "\~|~".

i think it will give absolutely same result but probably would be better as refactoring solution

Reply:
No, it won't give the same result.
Example: What if music's name will be contains '|'? Or '\~'?